Sources/GoodNetworking/DataRequestExtensions.swift

```diff
@@ -53,23 +55,24 @@ public extension DataRequest {
     ///   - queue:               `DispatchQueue` on which the `DataResponse` will be published. `.main` by default.
     ///   - preprocessor:        `DataPreprocessor` which filters the `Data` before serialization. `PassthroughPreprocessor()`
     ///                          by default.
-    ///   - decoder:             `DataDecoder` instance used to decode response `Data`. `JSONDecoder()` by default.
     ///   - emptyResponseCodes:  `Set<Int>` of HTTP status codes for which empty responses are allowed. `[204, 205]` by
     ///                          default.
     ///   - emptyRequestMethods: `Set<HTTPMethod>` of `HTTPMethod`s for which empty responses are allowed, regardless of
     ///                          status code. `[.head]` by default.
-    ///
+    ///   - decoder:             `JSONDecoder` instance used to decode response `Data`. For `Decodable` `JSONDecoder()` by default.
+    ///                          For `Decodable & WithCustomDecoder` custom `decoder` used by default.
     /// - Returns:               The `DataResponsePublisher`.
     func goodify<T: Decodable>(
         type: T.Type = T.self,
         queue: DispatchQueue = .main,
         preprocessor: DataPreprocessor = DecodableResponseSerializer<T>.defaultDataPreprocessor,
         emptyResponseCodes: Set<Int> = DecodableResponseSerializer<T>.defaultEmptyResponseCodes,
-        emptyResponseMethods: Set<HTTPMethod> = DecodableResponseSerializer<T>.defaultEmptyRequestMethods
+        emptyResponseMethods: Set<HTTPMethod> = DecodableResponseSerializer<T>.defaultEmptyRequestMethods,
+        decoder: JSONDecoder = (T.self as? WithCustomDecoder.Type)?.decoder ?? JSONDecoder()
     ) -> AnyPublisher<[T], AFError> {
         let serializer = DecodableResponseSerializer<[T]>(
             dataPreprocessor: preprocessor,
-            decoder: T.decoder,
+            decoder: JSONDecoder(),
```

Comment:
is `decoder` parameter required, when the implementation seems to create a new instance of `JSONDecoder` every time anyways?